### PR TITLE
Increased test coverage for forms.ImageField.to_python().

### PR DESCRIPTION
**Pull request by smithdc1**
Added a test for `ImageField` to check that an error is raised if Pillow doesn't recognise the image. This is to cover these lines [here](https://djangoci.com/view/%C2%ADCoverage/job/django-coverage/HTML_20Coverage_20Report/_home_jenkins_workspace_django-coverage_django_forms_fields_py.html#t649). 

The sample image I took from Pillow's own [repo](https://github.com/python-pillow/Pillow/blob/master/Tests/images/unimplemented_dxgi_format.dds), a little unsure if this is frowned upon? 

I tried to cover line 629/630 using `TemporaryUploadedFile`. Whilst I can get these lines to run by creating a 'fake' file, Pillow of course can't open it. So it didn't feel like a great test.

### CODE REVIEW

File: tests/forms_tests/field_tests/test_imagefield.py [hunk 1]
```diff
@@ -86,3 +86,14 @@ def test_widget_attrs_accept_false(self):
         f = ImageField(widget=FileInput(attrs={'accept': False}))
         self.assertEqual(f.widget_attrs(f.widget), {})
         self.assertWidgetRendersTo(f, '<input type="file" name="f" required id="id_f" />')
+
+    def test_unsupported_image_type(self):
+        f = ImageField()
+        img_path = get_img_path('filepath_test_files/unimplemented_dxgi_format.dds')
```
Comment (charettes) [inline tests/forms_tests/field_tests/test_imagefield.py:92]: Would using the Python test file for that to avoid adding a fixture work as well?

Comment (smithdc1) [reply tests/forms_tests/field_tests/test_imagefield.py]: Hi Simon - it works if I create a 'fake' image with `bytearray`. Have I interpreted your comment correctly?

Comment (charettes) [reply tests/forms_tests/field_tests/test_imagefield.py]: @smithdc1 it does thanks!

**Comment by felixxm** [comment]
@smithdc1 Thanks :+1: I added assertion also for "temporary" corrupted image.

**Comment by felixxm** [comment]
Merged in 911545da1d8d263f650721f6985e1d43e84d68eb.

**Pull request closed without merging**